plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveWithDisabledBatchFetchConnectorSmokeTest.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.thrift;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.plugin.hive.HiveConfig;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.testing.DataProviders;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static io.trino.plugin.hive.TestingThriftHiveMetastoreBuilder.testingThriftHiveMetastoreBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestHiveWithDisabledBatchFetchConnectorSmokeTest
+{
+    @Test(dataProviderClass = DataProviders.class, dataProvider = "trueFalse")
+    public void testSwitch(boolean enabled)
+    {
+        ThriftMetastore thriftMetastore = testingThriftHiveMetastoreBuilder()
```

Comment:
extract a method that creates that and then let's have two tests instead of using data provider and having "if" in test

Reply:
Extracted and split into two tests